### ISSUE TITLE
State feedback while selecting interesting seeds

### ISSUE THREAD

**Issue by mengrj** [aflnet/aflnet#101]
@thuanpv While checking whether a seed is interesting in the `save_if_interesting` function, one seed would be deemed uninteresting if it doesn't cover new branches (see [line 4001](https://github.com/aflnet/aflnet/blob/213c9cf1ecfabffee8b5771bb6966136e6cf05ba/afl-fuzz.c#L4001C2-L4001C2)) and the function is returned directly in [line 4003](https://github.com/aflnet/aflnet/blob/213c9cf1ecfabffee8b5771bb6966136e6cf05ba/afl-fuzz.c#L4003). However, the AFLNet paper mentions: 
```
A sequence is considered as interesting if the server response contains new states or state transitions that have not previously been observed (i.e., they are not recorded in the IPSM S); a sequence is interesting also if it covers new branches in the server’s source code
```
It seems the state feedback is not taken into consideration in the implementation. Are there any assumptions here?

**Comment by thuanpv** [comment]
Hi @mengrj . You are right. In our implementation, we did that because of two reasons. First, we assumed that new state behaviour is likely to trigger new branch(es) or increase hit counts, which might not be always true. Second, to be honest, I did not find a good solution to handle potentially much larger corpus (if we take into account state transitions) at that time. Due to that, we released the code that is a bit different from what we described in the paper. Our local comparisons did not show much differences between these two setups. I have ideas to address these issues in our next extension, which could be done in a journal submission of AFLNet. If you would like to contribute, I am more than happy to share my ideas with you. I have to postpone that plan several times due to other commitments.
